packages/core/admin/admin/src/content-manager/pages/ListView/components/BulkActionButtons/SelectedEntriesModal/index.js

```diff
@@ -128,26 +303,57 @@ const SelectedEntriesModalContent = ({ onToggle, onConfirm }) => {
 SelectedEntriesModalContent.propTypes = {
   onToggle: PropTypes.func.isRequired,
   onConfirm: PropTypes.func.isRequired,
+  onRefresh: PropTypes.func.isRequired,
 };
 
 /* -------------------------------------------------------------------------------------------------
  * SelectedEntriesModal
  * -----------------------------------------------------------------------------------------------*/
 
 const SelectedEntriesModal = ({ onToggle, onConfirm }) => {
-  const { rows, selectedEntries } = useTableContext();
+  const { selectedEntries, rows } = useTableContext();
+  const { contentType } = useSelector(listViewDomain());
 
-  // Get the selected entries full data, and keep the list view order
-  // Memoize to prevent infinite useEffect runs in SelectedEntriesTableContent
-  const entries = React.useMemo(() => {
-    return rows.filter((row) => {
-      return selectedEntries.includes(row.id);
-    });
-  }, [rows, selectedEntries]);
+  // We want to keep the selected entries order same as the list view
+  const [
+    {
+      query: { sort },
+    },
+  ] = useQueryParams();
+  const queryParams = {
+    sort,
+    filters: {
+      id: {
+        $in: selectedEntries,
+      },
+    },
+  };
+
+  const { get } = useFetchClient();
+  const queryString = stringify(queryParams);
+
+  const { data, isFetching, refetch } = useQuery(
+    ['entries', contentType.uid, queryParams],
+    async () => {
+      const { data } = await get(
+        `content-manager/collection-types/${contentType.uid}?${queryString}`
+      );
+
+      return data.results;
+    },
+    {
+      initialData: () => rows.filter((row) => selectedEntries.includes(row.id)),
+    }
+  );
 
   return (
-    <Table.Root rows={entries} defaultSelectedEntries={selectedEntries} colCount={4}>
-      <SelectedEntriesModalContent onToggle={onToggle} onConfirm={onConfirm} />
+    <Table.Root
+      rows={data}
+      defaultSelectedEntries={selectedEntries}
+      colCount={4}
+      isLoading={isFetching}
```

Comment:
lets change this prop to be a bit stricter e.g if there's existing data then the table doesn't show "loading" again and the refresh button could instead.